### PR TITLE
kubeadm: add test name output for RunInitNodeChecks and RunJoinNodeChecks

### PR DESCRIPTION
**Pull request by RobertKielty**
**What type of PR is this?**
/kind cleanup
/kind flake
**What this PR does / why we need it**:
Logs the name of the test case that caused a failure when [TestRunInitNodeChecks](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/preflight/checks_test.go#L187) runs
**Which issue(s) this PR fixes**:

This is intended to triage the flake reported on #93883

**Special notes for your reviewer**:
If you think that there are any other improvements that can be made to this test then I would be happy to implement them now.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
N/A

### CODE REVIEW

**Comment by neolit123** [comment]
/hold
we should instead move / remove / refactor these tests and they are not unit tests.
added similar comment here https://github.com/kubernetes/kubernetes/issues/93883

File: cmd/kubeadm/app/preflight/checks_test.go [hunk 1]
```diff
@@ -248,7 +248,8 @@ func TestRunInitNodeChecks(t *testing.T) {
 		actual := RunInitNodeChecks(exec.New(), rt.cfg, sets.NewString(), rt.isSecondaryControlPlane, rt.downloadCerts)
 		if (actual == nil) != rt.expected {
 			t.Errorf(
-				"failed RunInitNodeChecks:\n\texpected: %t\n\t  actual: %t\n\t error: %v",
+				"failed RunInitNodeChecks:%v\n\texpected: %t\n\t  actual: %t\n\t error: %v",
+				rt.name,
```
Comment (neolit123) [inline cmd/kubeadm/app/preflight/checks_test.go:252]: ok, i had a look at the level of refactor this requires in it's not that trivial.
we have code paths that are still tested here, so i'm going to log a ticket in k/kubeadm with a brief plan on how to proceed.

in the meantime we can merge this PR, but please add a similar "rt.name" in TestRunJoinNodeChecks.

Comment (RobertKielty) [reply cmd/kubeadm/app/preflight/checks_test.go:252]: Yes! Thanks so much, I was just about to fix that :-D  I'll fix the case with no name.

**Comment by neolit123** [comment]
logged: https://github.com/kubernetes/kubeadm/issues/2251

File: cmd/kubeadm/app/preflight/checks_test.go (hunk 1 above)
Comment (hasheddan) [reply cmd/kubeadm/app/preflight/checks_test.go:253]: @RobertKielty I think the ask here is to update https://github.com/kubernetes/kubernetes/pull/93885/files#diff-a54a9514dfa4dd1ab4cc9ada7445c91dR297 below with `rt.name` as well :slightly_smiling_face:

**Comment by neolit123** [comment]
/remove-kind flake
(given this PR adds test names)

/priority backlog

**Comment by neolit123** [comment]
/hold cancel
@RobertKielty 
please squash the commits to one.

File: cmd/kubeadm/app/preflight/checks_test.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/app/preflight/checks_test.go:253]: yes, please add  "rt.name" in TestRunJoinNodeChecks

Comment (RobertKielty) [reply cmd/kubeadm/app/preflight/checks_test.go:253]: I misread this feedback, will add the rt.name TestRunJoinNodeChecks now.

File: cmd/kubeadm/app/preflight/checks_test.go [hunk 2]
```diff
@@ -263,10 +265,12 @@ func TestRunJoinNodeChecks(t *testing.T) {
 		expected bool
```
Comment (hasheddan) [inline cmd/kubeadm/app/preflight/checks_test.go:265]: @RobertKielty need to add `name` up here

**Comment by neolit123** [comment]
/retitle kubeadm: adds test name output for RunInitNodeChecks and RunJoinNodeChecks

**Comment by neolit123** [comment]
there seems to be a gofmt failure:
```
 		{
-			name: "Check empty JoinConfiguration",
+			name:     "Check empty JoinConfiguration",
 			cfg:      &kubeadmapi.JoinConfiguration{},
 			expected: false,
 		},
```

you can run `./hack/update-gofmt.sh` and commit the changes.

**Comment by neolit123** [comment]
/retitle kubeadm: add test name output for RunInitNodeChecks and RunJoinNodeChecks

**Comment by neolit123** [comment]
@RobertKielty we should try to only apply critical fixes during code freeze.
/milestone clear

**Comment by RobertKielty** [comment]
@neolit123 Thank you for this.

**Pull request merged**